pkgs/stdenv/generic/setup.sh

```diff
@@ -1508,31 +1508,20 @@ distPhase() {
 
 showPhaseHeader() {
     local phase="$1"
-    case "$phase" in
-        unpackPhase) echo "unpacking sources";;
-        patchPhase) echo "patching sources";;
-        configurePhase) echo "configuring";;
-        buildPhase) echo "building";;
-        checkPhase) echo "running tests";;
-        installPhase) echo "installing";;
-        fixupPhase) echo "post-installation fixup";;
-        installCheckPhase) echo "running install tests";;
-        *) echo "$phase";;
-    esac
+        echo "started phase: $phase"
 }
 
 
 showPhaseFooter() {
     local phase="$1"
     local startTime="$2"
     local endTime="$3"
-    local delta=$(( endTime - startTime ))
-    (( delta < 30 )) && return
 
+    local delta=$(( endTime - startTime ))
     local H=$((delta/3600))
     local M=$((delta%3600/60))
     local S=$((delta%60))
-    echo -n "$phase completed in "
+    echo -n "finished phase: $phase in "
```

Comment:
Honestly do we need this always because it's very easy to tell when a phase ended with just the header

```
gnu-config> phases: unpackPhase patchPhase configurePhase buildPhase checkPhase installPhase fixupPhase installCheckPhase distPhase
gnu-config> started phase: unpackPhase
gnu-config> finished phase: unpackPhase in 0 seconds
gnu-config> started phase: patchPhase
gnu-config> finished phase: patchPhase in 0 seconds
gnu-config> started phase: installPhase
gnu-config> finished phase: installPhase in 0 seconds
gnu-config> started phase: fixupPhase
```

```
bison> phases: unpackPhase patchPhase updateAutotoolsGnuConfigScriptsPhase configurePhase buildPhase checkPhase installPhase fixupPhase installCheckPhase distPhase
bison> started phase: unpackPhase
bison> unpacking source archive /nix/store/9vm1ihdg1ysmrjdbb80g834iizzxb4yk-bison-3.8.2.tar.gz
bison> source root is bison-3.8.2
bison> setting SOURCE_DATE_EPOCH to timestamp 1632561040 of file bison-3.8.2/src/parse-gram.output
bison> finished phase: unpackPhase in 0 seconds
bison> started phase: patchPhase
bison> finished phase: patchPhase in 0 seconds
bison> started phase: updateAutotoolsGnuConfigScriptsPhase
bison> Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.sub
bison> Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.guess
bison> finished phase: updateAutotoolsGnuConfigScriptsPhase in 0 seconds
bison> started phase: configurePhase
```

Kinda cluttery

Reply:
If the phases are not doing anything, then this invites to set e.g. `dontUnpack = true;`, `dontPatch = true;`, `dontFixup = true;`...

Reply:
Often they're doing something but it's so fast and not logged that it's not visible

We have NIX_DEBUG=7 to see all the unimportant debug stuff

Some dontX should not be set without considerations because what if in the future something does something in that phase.

Reply:
Footer message might looks redundant, but what I don't like is it footer message is not printed consistently - it is printed only if phase execution takes longer time. I think I would like either dropping footer message completely or printing it all the time to avoid any confusion for the log reader.

Reply:
I wonder if footer message and header of next phase could be merged into one line?

Reply:
@maralorn , thank you for contributing to this discussion.

I personally, don't like this idea because it would complicate implementation little bit and can make navigation in the output little bit harder for the log reader. I see only little advantage of this which is reduction of few lines in the build output which is usually containing few hundreds or thousands of lines.

Reply:
Do we have consensus on this @Artturin @lorenzleutgeb @maralorn ?
Given that we have kinda 886G of build logs, I would be in favor of anything that contributes towards slowing down the rate of creating more bytes to the build logs.

Reply:
It seems that there's agreement on unconditionally printing the start of every phase, but that there is no agreement on printing the end of every phase (my preference is to unconditionally print it, but I personally don't have to deal with hundreds of gibibytes of build logs...). A compromise that I'd agree with would be to leave the logic for printing the end of a phase unchanged, merge this PR, and then continue a separate discussion towards https://github.com/NixOS/nixpkgs/pull/257229#issuecomment-1774259165.